protocol/nats_jetstream/v3/options.go

```diff
@@ -0,0 +1,110 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+const (
+	// field names
+	fieldURL              = "URL"
+	fieldConsumerConfig   = "ConsumerConfig"
+	fieldSendSubject      = "SendSubject"
+	fieldPullConsumerOpts = "PullConsumerOptions"
+	fieldPublishOptions   = "PublishOptions"
+	fieldFilterSubjects   = "FilterSubjects"
+
+	// error messages
+	messageNoConnection                 = "URL or nats connection must be given."
+	messageConflictingConnection        = "URL and nats connection were both given."
+	messageNoConsumerConfig             = "No consumer config was given."
+	messageNoFilterSubjects             = "No filter subjects were given."
+	messageMoreThanOneStream            = "More than one stream for given filter subjects."
+	messageNoSendSubject                = "Cannot send without a NATS subject defined."
+	messageMoreThanOneConsumerConfig    = "More than one consumer config given."
+	messageReceiverOptionsWithoutConfig = "Receiver options given without consumer config."
+	messageSenderOptionsWithoutSubject  = "Sender options given without send subject."
```

Comment:
nit: they're not used in this file. Suggesting to move all validation logic and errors to `validation.go` to keep the code easy to follow